docs/source/plugin.rst

```diff
@@ -100,6 +100,17 @@ A Sopel plugin consists of a Python module containing one or more
 Callable decorators
 ===================
 
+.. automodule:: sopel.plugin
+   :members:
+
+About ``sopel.module``
+----------------------
+
+Before Sopel 7.1, ``sopel.module`` was the preferred and only way to decorate
+callables for plugins. However, since ``module`` can be confusing (mostly
+because it already has a special meaning in Python), it has been replaced by
+``plugin`` in most case.
```

Comment:
```suggestion
Before Sopel 7.1, ``sopel.module`` was the preferred and only way to decorate
callables for plugins. However, since the term ``module`` can be confusing
(mostly because it already has a special meaning in Python), it has been
replaced by ``plugin`` in most cases related to add-ons for Sopel.
```

N.B.: I haven't checked how this renders, and am not sure if I'd rather use `"` quotes instead of monospace, but… I also honestly don't think it matters.